File: cc/020.cc
// Project Euler Problem 020
//
// Find the sum of digits in 100!

#include <cstdio>
#include <vector>
#include "bigmath.h"

using namespace std;

int main(int argc, const char* argv[]) {
  Bignum10 a(1);

  for (int i=2; i<=100; i++) {
    a *= i;
  }
  printf("%d\n", a.DigitSum());
}

File: go/020.go
// Project Euler Problem 020
//
// Find the sum of digits in 100!

package main

import (
	"big"
	"fmt"
)

func main() {
	s := big.NewInt(1).MulRange(2, 100).String()
	sum := 0
	for _, c := range s {
		sum += c - '0'
	}
	fmt.Println(sum)
}
